src/components/TablePagination/TablePagination.tsx

```diff
@@ -85,93 +45,145 @@ export type Props = PropsWithChildren<{
+    position = "above",
+    externallyControlled,
+    children,
+    ...divProps
+  } = props;
+
+  // Safety check to ensure pageSize is a valid option in
+  // pageLimits if the component is externally controlled
+  if (externallyControlled) {
+    let pageSizeFound = false;
+    for (const limit of pageLimits) {
+      if (limit === Number(props.pageSize)) {
+        pageSizeFound = true;
+        break;
+      }
+    }
+
+    if (!pageSizeFound) {
+      throw new Error(
+        `pageSize must be a valid option in pageLimits, pageLimits is set to [${pageLimits}]`
+      );
+    }
+  }
+
+  const [internalPageSize, setInternalPageSize] = useState(() => {
     return generatePagingOptions(pageLimits)[0].value;
   });
-  const { paginate, currentPage, pageData, totalItems } = usePagination(data, {
-    itemsPerPage: pageSize,
+  const {
+    paginate,
+    currentPage: internalCurrentPage,
+    pageData: internalData,
+  } = usePagination(externallyControlled ? [] : data, {
+    itemsPerPage: internalPageSize,
     autoResetPage: true,
   });
 
-  useEffect(() => {
-    const handleResize = () => {
-      setSmallScreen(figureSmallScreen(descriptionRef));
-    };
-    window.addEventListener("resize", handleResize);
-    return () => {
-      window.removeEventListener("resize", handleResize);
-    };
-  }, [isSmallScreen]);
+  const controlData = externallyControlled ? data : internalData;
+  const controlPageSize = externallyControlled
+    ? props.pageSize
+    : internalPageSize;
+  const controlTotalItems = externallyControlled
+    ? props.totalItems
+    : data.length;
+  const controlCurrentPage = externallyControlled
+    ? props.currentPage
+    : internalCurrentPage;
 
-  const handlePageSizeChange = (e: ChangeEvent<HTMLSelectElement>) => {
-    paginate(1);
-    setPageSize(parseInt(e.target.value));
-  };
-
-  const getDescription = () => {
-    if (description) {
-      return description;
+  const handlePageChange = (page: number) => {
+    if (externallyControlled) {
+      props.onPageChange(page);
+      return;
     }
 
-    const visibleCount = pageData.length;
-
-    if (isSmallScreen) {
-      return `${visibleCount} out of ${totalItems}`;
-    }
+    paginate(page);
+  };
 
-    if (visibleCount === totalItems && visibleCount > 1) {
-      return `Showing all ${totalItems} ${itemName}s`;
+  const handlePageSizeChange = (pageSize: number) => {
+    if (externallyControlled) {
+      props.onPageSizeChange(pageSize);
+      return;
     }
 
-    return `Showing ${visibleCount} out of ${totalItems} ${itemName}${
-      totalItems !== 1 ? "s" : ""
-    }`;
+    paginate(1);
+    setInternalPageSize(pageSize);
   };
 
-  const totalPages = Math.ceil(data.length / pageSize);
-  const clonedChildren = renderChildren(children, dataForwardProp, pageData);
+  const clonedChildren = renderChildren(children, dataForwardProp, controlData);
   return (
     <>
       {position === "below" && clonedChildren}
```

Comment:
Would it be easier to gasp if we assign a `const controls = <TablePaginationcontrols ...` above the return block, and change this to something like below?

```
return (
  <>
    {position === "above" && controls}
    {clonedChildren}
    {position === "below" && controls}
```

Comment

Reply:
I agree, this is better and no longer inverted!